### PR TITLE
Fix bugs to make CNN converted to SNN successfully

### PR DESCRIPTION
**Pull request by WeihaoTan**
modified: bindsnet/conversion/conversion.py
modified: bindsnet/network/network.py
modified: bindsnet/network/topology.py

Because I only run the CNN converted to SNN code for test. The change may be incompatible with other code and other tasks. More tests are needed. See comments to get the explaination of the revision.

### CODE REVIEW

File: bindsnet/network/topology.py [hunk 1]
```diff
@@ -419,7 +419,13 @@ def compute(self, s: torch.Tensor) -> torch.Tensor:
             return_indices=True,
         )
 
-        return s.take(indices).float()
+        B, C, H, W = indices.shape
+        output = torch.zeros_like(indices)
+        for b in range(B):
+            for c in range(C):
+                        output[b][c] = s[b][c].take(indices[b][c])
```
Comment (sharath) [inline bindsnet/network/topology.py]: indenting

**Review by sharath — APPROVED** [review]
I'm not really sure how the conversion works so I can't give much feedback on this.

**Comment by djsaunde** [comment]
Hey @WeihaoTan, this looks good. Thanks for catching the issues with the existing code. A few questions / comments:

1. Could you share the CNN -> SNN conversion code you're using? I'm interested to test it out for myself.
2. It would be good to remove the loops in the `MaxPool2DConnection.compute` function, but I think it's reasonable for now (especially since this version works).

I think we can get this merged pretty quickly.

**Comment by WeihaoTan** [comment]
Hi @djsaunde, here is my code. It doesn't support attaching py file. I change it to txt. 
[conversion_LeNet.txt](https://github.com/BINDS-LAB-UMASS/bindsnet/files/4282929/conversion_LeNet.txt)

**Comment by WeihaoTan** [comment]
> I'm not sure I understand this, but I'll try to look into it.
I think this may help you. It's the output of one loop of the original code.  There is one item called output in the file, which is the output of the s.take.

[maxpool_output.txt](https://github.com/BINDS-LAB-UMASS/bindsnet/files/4282954/maxpool_output.txt)

File: bindsnet/network/topology.py (hunk 1 above)
Comment (dee0512) [reply bindsnet/network/topology.py]: Also fix this.

**Comment by djsaunde** [comment]
Great work @WeihaoTan! Merging now.

**Pull request merged**